SYCL/Basic/alloc_on_subdevice.cpp

```diff
@@ -0,0 +1,32 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env CreateMultipleSubDevices=2 NEOReadDebugKeys=1 SYCL_DEVICE_FILTER="gpu" %t.out
+
+// REQUIRES: gpu
+
+#include <CL/sycl.hpp>
+
+#include <iostream>
+using namespace sycl;
+
+class KernelNameA;
+int main() {
+  device dev = device::get_devices(info::device_type::gpu)[0];
+  int val = 0;
+  std::vector<device> subdevices;
+
+  subdevices = dev.create_sub_devices<
+      info::partition_property::partition_by_affinity_domain>(
+      info::partition_affinity_domain::numa);
+  std::cout << "Testing on " << subdevices.size() << " subdevices" << std::endl;
+
+  context commonContext{subdevices};
+  for (int j = 0; j < subdevices.size(); ++j) {
+    buffer<int, 1> buf(&val, range<1>(1));

```

Comment:
Probably, this should be allocated outside the loop.

Reply:
It probably could be, but this would potentially change the tests behavior such that it no longer exhibits the issue for which it was created: https://github.com/intel/llvm/pull/4317. I'd prefer to keep E2E tests as is without trying to "improve" them.

Reply:
Got it.